### ISSUE TITLE
Spawn code/await within every

### ISSUE THREAD

**Issue by rodrimc** [fsantanna/ceu#85]
I'm not sure whether the compilation of this code should fail or not:
```
 code/await Test(void) -> void                                                   
 do                                                                              
   _printf ("tick\n");                                                           
 end                                                                             
                                                                                 
 every 1s do                                                                     
   spawn Test ();                                                                
 end 
```
Currently it's failing with the following message:
```
line 11 : invalid `await´ : unexpected enclosing `every´
```

If this is the expected behavior, could you please point me why? As far as I understand, a `spawn` statement spawns in parallel the `code`, which shouldn't interfere the `every` block.

Version: 2a2ab8e68ff31e811dde47e9f1bd3676c575b29a

**Comment by fsantanna** [comment]
Do you mean `spawn Test() in <some-pool>`?
If not, the spawned `Test` would be immediately aborted due to the `every` restart.
Do you have any use case for this behavior?

**Comment by rodrimc** [comment]
>Do you mean spawn Test() in <some-pool>?
>If not, the spawned Test would be immediately aborted due to the every restart.

Indeed I didn't spawned the `Test ()` code in a `pool`. However, regardless this (semantic) problem, I think the compilation shouldn't fail, don't you think?

>Do you have any use case for this behavior?

As a matter of fact I do. The following code excerpt spawns the `Send_Broadcast_Message` code1 to every message received.
```
 every message in new_message do                                
    spawn Send_Broadcast_Message (...);                              
 end
```

PS: This piece of code was compiling until some weeks ago.

**Comment by fsantanna** [comment]
It shouldn't compile because it doesn't make sense to have a `code/await` that is aborted immediately.
A `code/tight` makes more sense here.
What does `Send_Broadcast_Message` do?

**Comment by rodrimc** [comment]
`Send_Broadcast_Message` assembles a message and calls `UV_TCP_Write` to
send it through the socket.

>It shouldn't compile because it doesn't make sense to have
a code/await that is aborted immediately.

Note the error message doesn't seem to be related to this behaviour.

**Comment by fsantanna** [comment]
So, `UV_TCP_Write` will never complete.

The message could be better, for sure.
(The reason is that the `spawn` expands to a `par/or` + `await`.)

**Comment by rodrimc** [comment]
I see your point. However, it worth noting this code was compiling and working as I expected (that is, messages were properly sent) until some weeks ago.

If I spawn `Test() in <some-pool>` it's suppose to work? I mean, the code should compile and the `UV_TCP_Write` will complete (assuming the pool will live enough time for this operation have the chance to finish)?

**Comment by fsantanna** [comment]
Yes, spawning in a pool seems to be what you meant in the first place.
I guess It was working because you were sending a few bytes.

**Comment by rodrimc** [comment]
Ok, I see it now.

Do you have an example of spawning a code in a pool in the documentation?

**Comment by fsantanna** [comment]
Not yet.

```
pool[] Xx xs;
...
    spawn Xx(...) in xs;
```

**Comment by rodrimc** [comment]
Thanks.
